File: solutions/fsharp/series/1/Series.fs
module Series

let slices (str:string) (length:int) : string list Option =

    match (length > String.length str, length < 1) with
    | (true, _) -> None
    | (_, true) -> None
    | _ ->
        Some [ for idx in [ 0 .. String.length str - 1 ] do
                   if Seq.length (str[idx ..]) >= length then
                       yield str[idx .. idx + length - 1] ]

File: solutions/csharp/series/2/Series.cs
using System.Linq;

public static class Series
{
    public static string[] Slices(string nums, int length)
    {
        if (length > nums.Length || length < 1 || nums == "")
            throw new ArgumentException("NoNo");
        else
        {
            IEnumerable<string> Loop()
            {
                foreach (var idx in nums.Select((_, idx) => idx))
                {
                    var next = nums.Skip(idx).Take(length).ToArray();
                    if (next.Count() == length) yield return new string(next);
                    else break;
                }
            }

            return Loop().ToArray();
        }
    }
}